dropwizard-oauth/src/test/groovy/smartthings/dw/oauth/scope/ScopesAllowedDynamicFeatureSpec.groovy

```diff
@@ -242,6 +242,23 @@ class ScopesAllowedDynamicFeatureSpec extends Specification {
         'staticScope'        | StaticScope        | true       | 1           | 1         | 0
     }
 
+    @Unroll
+    def 'fineGrainedScopePartsMatch should work correctly with *s'() {
```

Comment:
Is there a test elsewhere in this file making sure we didn't regress on the `r:security:123-123-1234:disarm` case you mentioned? If not, probably should add that to this test.